manifests/agent.pp

```diff
@@ -334,14 +356,26 @@
   }
 
   # Include dir for specific zabbix-agent checks.
-  file { $include_dir:
-    ensure  => directory,
-    owner   => 'zabbix',
-    group   => 'zabbix',
-    recurse => true,
-    purge   => $include_dir_purge,
-    notify  => Service['zabbix-agent'],
-    require => File[$agent_configfile_path],
+  if $directory_ensure == 'directory' {
+    file { $include_dir:
+      ensure  => $directory_ensure,
+      owner   => 'zabbix',
+      group   => 'zabbix',
+      recurse => true,
+      purge   => $include_dir_purge,
+      notify  => Service['zabbix-agent'],
```

Comment:
what about creating a $purge variable which is undef or $include_dir_purge and don't duplicate the file resource? You could add that in the upper case block.